Classes/common/query/CDTQQuerySqlTranslator.m

```diff
@@ -85,27 +85,20 @@ + (CDTQQueryNode *)translateQuery:(NSDictionary *)query
         return nil;
     } else if (!state.textIndexRequired &&
                   (!state.atLeastOneIndexUsed || state.atLeastOneORIndexMissing)) {
-        // If we haven't used a single index, we need to return a query
-        // which returns every document, so the posthoc matcher can
+        // If we haven't used a single index or an OR clause is missing an index,
+        // we need to return every document id, so that the post-hoc matcher can
         // run over every document to manually carry out the query.
+        CDTQSqlQueryNode *sqlNode = [[CDTQSqlQueryNode alloc] init];
         NSSet *neededFields = [NSSet setWithObject:@"_id"];
-        NSString *allDocsIndex =
-            [CDTQQuerySqlTranslator chooseIndexForFields:neededFields fromIndexes:indexes];
+        NSString *allDocsIndex = [CDTQQuerySqlTranslator chooseIndexForFields:neededFields
+                                                                  fromIndexes:indexes];
 
-        if (!allDocsIndex) {
-            LogError(@"No indexes defined, cannot execute query for all documents");
-            return nil;
+        if (allDocsIndex.length > 0) {
```

Comment:
I'd be concerned if we didn't get into this block. This should be somewhat guaranteed right? There's always an index for the "_id" key, right?

If this is true, and allDocsIndex.length == 0 for some crazy, what should be done? Should it send a Log message and bail?